TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -76,22 +76,22 @@ class MetAPI{
             }
     }
     
-    func object(id: ArtID, completion: @escaping (ObjectResponse?) -> Void) {
+    func object(id: ArtID, completion: @escaping (Result<ObjectResponse, MetAPIError>) -> Void) {
         self.metAPICache.object(id: id) { [weak self] objectResponce in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIdoesNotAnswer))
                 return
             }
             if let objectResponce = objectResponce {
-                completion(objectResponce)
+                completion(.success(objectResponce))
             } else {
                 self.executeObject(id: id) {[weak self] realResponse in
                     guard let realResponse = realResponse else {
-                        completion(nil)
+                        completion(.failure(.noDataInResponce))
```

Comment:
You still rely on `nil`/`non-nil` state of the `realResponse`. Please introduce `Result` to `func executeObject` as well